1-js/04-object-basics/06-constructor-new/article.md

```diff
@@ -60,106 +60,105 @@ let user = {
-  if (!new.target) { // if you run me without new
-    return new User(name); // ...I will add new for you
+  if (!new.target) { // se você me executar sem new
+    return new User(name); // ...Eu irei adicioná-lo para você
   }
 
   this.name = name;
 }
 
-let john = User("John"); // redirects call to new User
+let john = User("John"); // redireciona a chamada para new User
 alert(john.name); // John
 ```
 
-This approach is sometimes used in libraries to make the syntax more flexible. So that people may call the function with or without `new`, and it still works.
+Essa abordagem é utilizada algumas vezes em bibliotecas para deixar a sintaxe mais flexível. Para que as pessoas possam chamar as funções com ou sem `new`, e ela ainda sim, funcionará.
+Porém, provavelmente não é uma boa usar em qualquer lugar, pois omitir `new` faz ficar um pouco menos óbvio o que está acontecendo. Com `new` todos sabemos que o novo objeto está sendo criado.
 
-Probably not a good thing to use everywhere though, because omitting `new` makes it a bit less obvious what's going on. With `new` we all know that the new object is being created.
+## Return em construtores
 
-## Return from constructors
+Geralmente, construtores não têm uma declaração `return`. A tarefa deles é escrever todas as coisas dentro do `this`, e ele automaticamente se torna o resultado.
 
-Usually, constructors do not have a `return` statement. Their task is to write all necessary stuff into `this`, and it automatically becomes the result.
+Mas se existir uma declaração `return`, então a regra é simples:
 
-But if there is a `return` statement, then the rule is simple:
+- Se `return` é chamado com um objeto, então ele é retornado ao invés de `this`.
+- Se `return` é chamado com um primitivo, ele é ignorado.
 
-- If `return` is called with object, then it is returned instead of `this`.
-- If `return` is called with a primitive, it's ignored.
+Em outras palavras, `return` com um objeto retorna o objeto, em todos os outros casos, o `this` é retornado.
 
-In other words, `return` with an object returns that object, in all other cases `this` is returned.
-
-For instance, here `return` overrides `this` by returning an object:
+Por exemplo, aqui, `return` sobrepõe o `this` retornando um objeto:
 
 ```js run
 function BigUser() {
 
   this.name = "John";
 
-  return { name: "Godzilla" };  // <-- returns an object
+  return { name: "Godzilla" };  // <-- retorna um objeto
 }
 
-alert( new BigUser().name );  // Godzilla, got that object ^^
+alert( new BigUser().name );  // Godzilla, pegou aquele objeto ^^
 ```
 
-And here's an example with an empty `return` (or we could place a primitive after it, doesn't matter):
+E aqui está um exemplo com um `return` vazio (ou poderíamos colocar um primitivo depois dele, não importa):
 
 ```js run
 function SmallUser() {
 
   this.name = "John";
 
-  return; // finishes the execution, returns this
+  return; // termina a execução, retorna this
 
   // ...
 
```

Comment:
- code snippet has changed, available at [Return from constructors](https://javascript.info/constructor-new#return-from-constructors) (2nd block)
("// ..." needs to be replaced by the function closing brace)